homeassistant/components/knx/__init__.py

```diff
@@ -80,6 +80,9 @@
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.All(
+            # deprecated since 2021.3
+            cv.deprecated(CONF_KNX_CONFIG, replacement_key=CONF_KNX_CONFIG),
```

Comment:
The 'CONF_KNX_CONFIG' option is deprecated, please replace it with 'CONF_KNX_CONFIG'

This doesn't look right that the `replacement_key` is the same

Reply:
This doesn't really break things, it will fire a warning from the cv.deprecated config validator for now.
If I don't specify a `replacement_key` the config validator removes the key from config and it can not be used further down. This would result in users loosing some configuration like HomeKit room assignment (afaik).

So for now I'd like to just point out the deprecation of this config and log a link to the config-converter. After a transition period is over (maybe in summer) well remove the `replacement_key` and the logic using the XKNX-Schema. 

I was more worried about the link to an external resource (config-converter tool) being accepted.

Reply:
would it be ok to also create a `persistent_notification` ?

Reply:
I'm not sure about an external link in a notification. I'll ask for a second opinion on that.

Reply:
If it is any help: xknx.io is also the only source for documentation of the deprecated XKNX-Schema - so users of this are probably familiar with this domain.

Reply:
Deprecated should not remove the key if you don't specify replacement. You have to explicitly remove it with a vol.Remove() if you want it removed.

Reply:
Also we don't generally do notifications on deprecated. If we want to do that i think we should do that globally.

Reply:
I've removed the deprecation config validator. It's a log normal log message and notification now. (Only when config_file is used). 
Shall I remove the notification again? I think a log message can be quite invisible for users - and this change will probably render their whole installation useless (knx is usually a quite big installation). When it is deprecated there is no way to point to the converter website...

Reply:
We should remove the external domain in the notification. If this is useful, add it to the documentation that is already linked instead.

Reply:
TBH I'm not really sure users will be pleased and/or successful in finding that link in this 1,3k lines markdown document.